[Java] 3.68.0 Release Notes (#809)
SAP/cloud-sdk: docs/java/release-notes.mdx

### Fixed Issues
-->

## 3.68.0 - May 19, 2022

[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.68.0) | [Javadoc](https://help.sap.com/doc/d36e6a93f11b48bd920453bed2149bd3/1.0/en-US/index.html)

### Compatibility Notes

- To fix the high vulnerability [CVE-2022-22968](https://nvd.nist.gov/vuln/detail/CVE-2022-22968) in Spring, we had to update [Spring BOM](https://search.maven.org/search?q=g:org.springframework%2BAND%2Ba:spring-framework-bom) to `5.3.19` and to maintain compatibility had to also update [XSUAA Client Library](https://github.com/SAP/cloud-security-xsuaa-integration) to `2.11.16`.
  Please note that if you are using SAP JAVA buildpack for your deployments use a static buildpack version `sap_java_buildpack_1_52` to avoid dependency conflicts and class loading issues.

> - Update the SAP Business Rules OpenAPI client library to the latest version of the Business Rules service.
>   This includes a breaking change to the field `List<TextPredefinedResults> predefinedResults` which is renamed to `List<TextPredefined> predefined`.

Reviewer: ```suggestion
-  SAP Buisiness Rules OpenAPI client library received breaking change to the field `List<TextPredefinedResults> predefinedResults` which is renamed to `List<TextPredefined> predefined` after updating to the latest version of the SAP Business Rules service
```

- Deprecate non-productive `com.sap.cloud.sdk.s4hana.datamodel.odata.adapter.ODataCalendarAdapter` and child classes `ODataDateTimeAdapter`, `ODataDateTimeOffsetAdapter`, and `ODataTimeAdapter`.
  It is planned to remove those classes with [SAP Cloud SDK v4](https://sap.github.io/cloud-sdk/docs/java/release-policy).
- Switch from using `com.google.code.findbugs:annotations` to `com.github.spotbugs:spotbugs-annotations` in modules `cloudplatform-core-dwc-cf` and `recast-ai`.
  No code changes are required if you are consuming these modules.
  However, if you are planning on consuming these dependencies, remember to declare them directly in your POM as these are declared as [optional dependencies](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html).
- The `DefaultLocaleFacade` does no longer consider the `ServletRequest` (accessed via the `RequestAccessor`) to determine the desired `Locale`.
  Instead, the facade uses the `RequestHeaderAccessor` to derive the `Locale` based on the `Accept-Language` header - this change was already introduced in version `3.61.0`.

>   As a fallback, in case the header cannot be found, the default `Locale` for the server will be returned - this has also not been changed.

Reviewer: ```suggestion
  As a fallback, in case the header cannot be found, the default `Locale` for the server will be returned as before.
```

  This change does not affect any productive scenarios.
  However, there might be changes in tests, where the `RequestServlet#getLocales()` is mocked to return an arbitrary value that is neither part of the `Accept-Language` header nor the default system `Locale`.
- Deprecate modules:
  - `Recast AI`
  - `Graph`
  - `Services - SAP BTP Machine Learning`

### Improvements

> - Update the SAP Business Rules OpenAPI client library to the latest version of the Business Rules service.

Reviewer: ```suggestion
- The SAP Business Rules OpenAPI client library was updated to the latest version of the Business Rules service.
```

        - Update [Rest Assured](https://search.maven.org/search?q=g:io.rest-assured%2BAND%2Ba:rest-assured) (`io.rest-assured:rest-assured`) from `5.0.0` to `5.0.1`
        - Update [Wiremock](https://search.maven.org/search?q=g:com.github.tomakehurst%2BAND%2Ba:wiremock-jre8-standalone) (`com.github.tomakehurst:wiremock-jre8-standalone`) from `2.32.0` to `2.33.2`
        - Update [Spring BOM](https://search.maven.org/search?q=g:org.springframework%2BAND%2Ba:spring-framework-bom) (`org.springframework:spring-framework-bom`) from `5.3.18` to `5.3.19`
        - Update [Spring Security](https://search.maven.org/search?q=g:org.springframework%2BAND%2Ba:spring-security-bom) (`org.springframework:spring-security-bom`) from `5.6.2` to `5.6.3`
        - Update [Jsonschema2Pojo Core](https://search.maven.org/search?q=g:org.jsonschema2pojo%2BAND%2Ba:jsonschema2pojo-core) (`org.jsonschema2pojo:jsonschema2pojo-core`) from `1.1.1` to `1.1.2`
        - Update [Togglz Core](https://search.maven.org/search?q=g:org.togglz%2BAND%2Ba:togglz-core) (`org.togglz:togglz-core`) from `3.1.1` to `3.1.2`

### Fixed Issues

> - Fix an issue with OpenAPI client not being able to instantiate the class `ApiClient` when _Jackson_ formats other than _JSON_ are possible at runtime.

Reviewer: ```suggestion
- Fix an issue with the OpenAPI client not being able to instantiate the class `ApiClient` when _Jackson_ formats other than _JSON_ are possible at runtime.
```
